app/src/main/java/com/lightricks/feedexercise/database/FeedDatabase.kt

```diff
@@ -1,5 +1,31 @@
 package com.lightricks.feedexercise.database
 
-/**
- * todo: add the abstract class that extents RoomDatabase here
- */
+import android.content.Context
+import android.os.strictmode.InstanceCountViolation
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+
+
+@Database(entities = [Entity::class], version = 1)
+abstract class FeedDatabase : RoomDatabase() {
+
+    abstract fun feedDao(): FeedDao
+
+    companion object {
+        private var INSTANCE: FeedDatabase? = null
+        fun getExistingDatabase(): FeedDatabase{
+            return INSTANCE as FeedDatabase
+        }
+        fun getDatabase(context: Context): FeedDatabase {
+            if (INSTANCE == null) {
```

Comment:
This is better. Now there is only one database object created.
In a real app you must also synchronize this block, because multiple thread could have a race creating the instance.